### ISSUE TITLE
[explicit-module-boundary-types] Conflicting errors about constructor return types

### ISSUE THREAD

**Issue by kaiyoma** [typescript-eslint/typescript-eslint#2172]
**Repro**

```TS
// https://github.com/MQuy/webpack-deadcode-plugin
declare module 'webpack-deadcode-plugin' {
  import { compilation, Compiler } from 'webpack';

  interface Options {
    patterns?: string[];
    exclude?: string[];
    context?: string;
    failOnHint?: boolean;
    outputFile?: string;
    detectUnusedFiles?: boolean;
    detectUnusedExport?: boolean;
  }

  export default class WebpackDeadcodePlugin {
    public constructor(options?: Options);

    public apply(compiler: Compiler): void;

    public handleAfterEmit(
      options: Options,
      compilation: compilation.Compilation,
      callback: () => void,
    ): void;
  }
}

```

**Additional Info**

If I run this code through eslint, I get this error:

```
16:23  warning  Missing return type on function  @typescript-eslint/explicit-module-boundary-types
```

If I add a return type, then run this code through the type checker, I get this error:

```
types/webpack-deadcode-plugin.d.ts:16:44 - error TS1093: Type annotation cannot appear on a constructor declaration.

16     public constructor(options?: Options): WebpackDeadcodePlugin;
                                              ~~~~~~~~~~~~~~~~~~~~~
```

Sooo... which one is it?

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.1.0` |
| `@typescript-eslint/parser`        | `3.1.0` |
| `TypeScript`                       | `3.9.2` |
| `ESLint`                           | `7.1.0` |
| `node`                             | `12.16.1` |
| `npm`                              | `6.13.4` |

**Comment by bradzacher** [comment]
#2150, #2163

Please use the search next time

**Comment by kaiyoma** [comment]
@bradzacher I did find those issues via searching, but I don't see how they apply here. The comments in the other issues talk about untyped parameters, which is not the case here:

```
public constructor(options?: Options);
```

Someone else also reported the same issue yesterday, but no one answered that either:

https://github.com/typescript-eslint/typescript-eslint/issues/2150#issuecomment-637764358

Based on what I'm reading, this is a new issue that's not addressed by the other ones.

**Comment by bradzacher** [comment]
#2150:
> [explicit-module-boundary-types] Warns on constructor overload
>
> **Expected Result**
> No warnings
>
> **Actual Result**
> Missing return type on function

Which is exactly this issue?
And the attached PR to that issue fixes it.

**Comment by kaiyoma** [comment]
Ah, I'm sorry, I thought that fix was in the latest version (3.1.0), but it appears that it isn't, which is why I'm still seeing the error. I'll try this again after the next version comes out. Thanks!

**Comment by kaiyoma** [comment]
@bradzacher When can we expect the next release (that has this fix)?

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint#package-versions

**Comment by kaiyoma** [comment]
Got it. The latest canary version fixes the constructor issues I was having. Thanks!
